### PR TITLE
Added Unit Test & Small fix

### PR DESCRIPTION
**Pull request by KevenLYF**
- Added Unit test for get student completed courses
    - fix
	1. Refactored some codes (e.g removed unnecessary async/await call)
	1. Some test cases do not release the connection

### CODE REVIEW

File: backend/logic/users/users.js [hunk 1]
```diff
@@ -29,8 +29,15 @@ var insertStudentUser = async (username, password, email, id) => {
 
   try {
     await connection.beginTransaction();
-    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [username, email, hash]);
-    await connection.query("INSERT INTO students VALUES(?, ?);", [id, username]);
+    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [
```
Comment (zerfuth) [inline backend/logic/users/users.js]:
```suggestion
    await connection.query("INSERT INTO users (username, email, password) VALUES(?, ?, ?);", [
```

File: backend/logic/users/users.js [hunk 2]
```diff
@@ -29,8 +29,15 @@ var insertStudentUser = async (username, password, email, id) => {
...
+      username,
+      email,
+      hash
+    ]);
+    await connection.query("INSERT INTO students VALUES(?, ?);", [
```
Comment (zerfuth) [inline backend/logic/users/users.js]:
```suggestion
    await connection.query("INSERT INTO students (student_id, username) VALUES(?, ?);", [
```

File: backend/logic/users/users.js [hunk 3]
```diff
@@ -99,8 +111,15 @@ var insertAdminUser = async (username, password, email, id) => {
   let hash = hasher.hashPass(password);
   try {
     await connection.beginTransaction();
-    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [username, email, hash]);
-    await connection.query("INSERT INTO staff_members VALUES(?, ?);", [id, username]);
+    await connection.query("INSERT INTO users VALUES(?, ?, ?);", [
```
Comment (zerfuth) [inline backend/logic/users/users.js]:
```suggestion
    await connection.query("INSERT INTO users (username, email, password) VALUES(?, ?, ?);", [
```

File: backend/logic/users/users.js [hunk 4]
```diff
@@ -99,8 +111,15 @@ var insertAdminUser = async (username, password, email, id) => {
...
+      username,
+      email,
+      hash
+    ]);
+    await connection.query("INSERT INTO staff_members VALUES(?, ?);", [
```
Comment (zerfuth) [inline backend/logic/users/users.js]:
```suggestion
    await connection.query("INSERT INTO staff_members (staff_id, username) VALUES(?, ?);", [
```

File: backend/test/users_test.js [hunk 5]
```diff
@@ -660,3 +663,64 @@ describe("Test retrieve student data", function() {
...
+      "getCompletedCourseTest",
+      "getCompletedCourseTest",
+      "getCompletedCourseTest@email.com",
+      260561054
+    );
+
+    const conn = await mysql.getNewConnection();
+    await conn.query(
+      `INSERT INTO student_course_offerings (student_id, offering_id, semester)
+    VALUES(?, ?, ?)`,
```
Comment (zerfuth) [inline backend/test/users_test.js]:
```suggestion
    VALUES(?, ?, ?);`,
```

File: backend/test/users_test.js [hunk 6]
```diff
@@ -660,3 +663,64 @@ describe("Test retrieve student data", function() {
...
+    );
+
+    await conn.release();
+  });
+
+  // clean up test data
+  after(async () => {
+    const conn = await mysql.getNewConnection();
+    await conn.query(
+      `DELETE FROM student_course_offerings WHERE student_id = ?`,
```
Comment (zerfuth) [inline backend/test/users_test.js]:
```suggestion
      `DELETE FROM student_course_offerings WHERE student_id = ?;`,
```

File: backend/test/users_test.js [hunk 7]
```diff
@@ -660,3 +663,64 @@ describe("Test retrieve student data", function() {
...
+      [260561054]
+    );
+    await conn.query(`DELETE FROM course_offerings WHERE id = ?`, [940915]);
```
Comment (zerfuth) [inline backend/test/users_test.js]:
```suggestion
    await conn.query(`DELETE FROM course_offerings WHERE id = ?;`, [940915]);
```

File: backend/test/users_test.js [hunk 8]
```diff
@@ -660,3 +663,64 @@ describe("Test retrieve student data", function() {
...
+    await conn.query(`DELETE FROM courses WHERE course_code = ?`, ["TEST 001"]);
```
Comment (zerfuth) [inline backend/test/users_test.js]:
```suggestion
    await conn.query(`DELETE FROM courses WHERE course_code = ?;`, ["TEST 001"]);
```

**Comment by KevenLYF** [comment]
@zerfuth  Good comments, don't know why I forgot to put fields after table name, like I did it for some of the queries. Gonna upload a new patch

**Comment by KevenLYF** [comment]
@zerfuth  Ohh wait actually I didn't wrote the code for `insertStudentUser` function in logic/users/users.js  
It was just auto-formatting lol, so I think I shouldn't touch it.
But I'm gonna add the semicolon you pointed out in `user_test.js`

**Comment by zerfuth** [comment]
I think it's fine if you fix it but in any case, i'll mark as good

**Comment by Ty-Won** [comment]
@KevenLYF nice work, please review the conflicts, once they are resolved, I'll merge this

**Comment by KevenLYF** [comment]
> @KevenLYF nice work, please review the conflicts, once they are resolved, I'll merge this

Thanks I'll fix the conflicts

File: backend/logic/courses/courses.js [hunk 9]
```diff
@@ -315,7 +315,7 @@ var updateCourse = async (course, newTitle, newTags) => {
  */
 let phaseOutCourse = async courseCode => {
   let connection = await mysql.getNewConnection();
-  await format.verifyCourseCode(courseCode);
+  format.verifyCourseCode(courseCode);
```
[the remark this answers is not in the document]
Comment (KevenLYF) [reply backend/logic/courses/courses.js]: I checked the method and it does not contain any asynchronous operations, so I think await is redundant here

**Pull request merged**